be/src/storage/olap_runtime_range_pruner.hpp

```diff
@@ -0,0 +1,126 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+#pragma once
+
+#include <memory>
+
+#include "exec/olap_common.h"
+#include "exprs/vectorized/runtime_filter_bank.h"
+#include "storage/olap_runtime_range_pruner.h"
+#include "storage/predicate_parser.h"
+#include "storage/vectorized_column_predicate.h"
+
+namespace starrocks::vectorized {
+namespace detail {
+struct RuntimeColumnPredicateBuilder {
+    template <PrimitiveType ptype>
+    StatusOr<std::vector<std::unique_ptr<ColumnPredicate>>> operator()(PredicateParser* parser,
+                                                                       const RuntimeFilterProbeDescriptor* desc,
+                                                                       const SlotDescriptor* slot) {
+        if constexpr (ptype == TYPE_TIME || ptype == TYPE_NULL || ptype == TYPE_JSON || pt_is_float<ptype>) {
+            CHECK(false) << "unreachable path";
```

Comment:
These types can not be pushed down because these types cannot be used as `short keys`?